Services/Form/classes/class.ilPropertyFormGUI.php

```diff
@@ -487,6 +488,13 @@ function checkInput()
 		{
 			ilUtil::sendFailure($lng->txt("form_input_not_valid"));
 		}
+		global $DIC;
+		if(!$ok && $DIC->http()->request()->getQueryParams()[ilFileStandardDropzoneInputGUI::ASYNC_FILEUPLOAD] ) {
+			echo json_encode([
+				'success'      => false,
+				'message'      => 'There was an error checking your form',
+			]);
+		}
```

Comment:
The general ilPropertyFormGUI class should not have a dependency on ilFileStandardDropzoneInputGUI. I am not sure how the whole processing works. In case of an error you want a JSON format, what happens in case of no error? Who calls checkInput? Your consuming code directly? The I would prefer something like checkInput($a_json_response = false), and you set this to true. Or a general "JSON response" mode in the form being triggered by a request parameter not specific to file upload (something like $_GET["cmdMode"] == "asynch" in ilCtrl).